modules/data-streams/src/main/java/org/elasticsearch/datastreams/lifecycle/downsampling/DeleteSourceAndAddDownsampleToDS.java

```diff
@@ -107,13 +112,13 @@ ClusterState execute(ClusterState state) {
                 throw new IllegalStateException(errorMessage);
             }
 
-            IndexMetadata sourceIndexMeta = state.metadata().getProject().index(sourceBackingIndex);
+            IndexMetadata sourceIndexMeta = project.index(sourceBackingIndex);
             assert sourceIndexMeta != null
                 : "the source index abstraction exists in the indices lookup, so the index metadata must "
                     + "exist in the same cluster state metadata";
             // the source index exists so let's start by deleting it
             state = MetadataDeleteIndexService.deleteIndices(state, Set.of(sourceIndexMeta.getIndex()), settings);
```

Comment:
I think we should pass project state here to use the other variant of `deleteIndices`. It feels more correct to me that we are definitely deleting from a single project.

Reply:
Good point! Done in [e0d85be](https://github.com/elastic/elasticsearch/pull/124808/commits/e0d85be6609188ace8a4e8be2c0b58780f64b4c5)